src/model/transaction/__tests__/insertFragmentIntoContentState-test.js

```diff
@@ -107,3 +118,277 @@ test('must apply multiblock fragments', () => {
     ]),
   );
 });
+
+test('must apply a single ContentBlockNode from fragment', () => {
+  const initialSelection = SelectionState.createEmpty('A');
+  const initialContent = contentState.set(
+    'blockMap',
+    createContentBlockNodeFragment([
+      {
+        key: 'A',
+        text: '',
+        data: null,
+      },
+    ]),
+  );
+
+  assertInsertFragmentIntoContentState(
+    createContentBlockNodeFragment([
+      {
+        key: 'B',
+        text: 'some text',
+      },
+    ]),
+    initialSelection,
+    initialContent,
+  );
+});
+
+test('must apply ContentBlockNode fragments', () => {
+  const initialSelection = SelectionState.createEmpty('first');
+  const initialContent = contentState.set(
+    'blockMap',
+    createContentBlockNodeFragment([
+      {
+        key: 'first',
+        text: '',
+        nextSibling: 'second',
+      },
+      {
+        key: 'second',
+        text: '',
+        prevSibling: 'first',
+      },
+    ]),
+  );
+
+  assertInsertFragmentIntoContentState(
+    createContentBlockNodeFragment([
+      {
+        key: 'B',
+        text: '',
+        children: List(['C']),
+        nextSibling: 'E',
+      },
+      {
+        key: 'C',
+        parent: 'B',
+        text: '',
+        children: List(['D']),
+      },
+      {
+        key: 'D',
+        parent: 'C',
+        text: 'Delta',
+      },
+      {
+        key: 'E',
+        text: 'Delta',
+        prevSibling: 'B',
+      },
+    ]),
+    initialSelection,
+    initialContent,
+  );
+});
+
+test('must apply ContentBlockNode fragments along side nested block', () => {
```

Comment:
Lets rename this to something more meaningful like,

`must be able to insert fragment of ContentBlockNodes after nested block`